File: src/main/resources/client-angular/src/app/file-entity.ts
export class FileEntity {

  private _id:number;
  private _content: String;
  private _fileName: string;
  private _fileAuthor: string;
  private _fileType: string;
  private _programToRun : string;


  constructor(id: number, file: String, fileName: string, fileAuthor: string, fileType: string, programToRun: string) {
    this._id = id;
    this._content = file;
    this._fileName = fileName;
    this._fileAuthor = fileAuthor;
    this._fileType = fileType;
    this._programToRun = programToRun;
  }

  get programToRun(): string {
    return this._programToRun;
  }

  set programToRun(value: string) {
    this._programToRun = value;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get file(): String {
    return this._content;
  }

  set file(value: String) {
    this._content = value;
  }

  get fileName(): string {
    return this._fileName;
  }

  set fileName(value: string) {
    this._fileName = value;
  }

  get fileAuthor(): string {
    return this._fileAuthor;
  }

  set fileAuthor(value: string) {
    this._fileAuthor = value;
  }

  get fileType(): string {
    return this._fileType;
  }

  set fileType(value: string) {
    this._fileType = value;
  }

}

File: src/main/java/com/targetlabsassignment/springfileupload/FileEntity.java
package com.targetlabsassignment.springfileupload;

import java.sql.Blob;
import java.util.Arrays;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class FileEntity {
	
	@Id
	@GeneratedValue
	int id;
	
	// meta data
	private String fileName;
	private String fileAuthor;
	private String fileType;
	private String programToRun;
	
	public FileEntity() {
		super();
	}

	public FileEntity(String fileName, String fileAuthor, String fileType, String programToRun) {
		super();
		this.fileName = fileName;
		this.fileAuthor = fileAuthor;
		this.fileType = fileType;
		this.programToRun = programToRun;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getFileName() {
		return fileName;
	}

	public void setFileName(String fileName) {
		this.fileName = fileName;
	}

	public String getFileAuthor() {
		return fileAuthor;
	}

	public void setFileAuthor(String fileAuthor) {
		this.fileAuthor = fileAuthor;
	}

	public String getFileType() {
		return fileType;
	}

	public void setFileType(String fileType) {
		this.fileType = fileType;
	}

	public String getProgramToRun() {
		return programToRun;
	}

	public void setProgramToRun(String programToRun) {
		this.programToRun = programToRun;
	}

	@Override
	public String toString() {
		return "FileEntity{" +
				"id=" + id +
				", fileName='" + fileName + '\'' +
				", fileAuthor='" + fileAuthor + '\'' +
				", fileType='" + fileType + '\'' +
				", programToRun='" + programToRun + '\'' +
				'}';
	}
}
